### PR TITLE
Explicitly document hardware and software support.

### LINKED ISSUE #1169: HOOMD restricted to CUDA compute capability >=6
## Description

I was surprised to find that in jumping from HOOMD-blue 2 to 3, the CUDA compute capability requirement has also been stepped up. At home I'm able to test on a 20 series card (where my code worked), but at my university we have a handful of old but still functional K80s to launch jobs to. Is there a technical reason for this (HOOMD now uses features from CC 6 that aren't available in later generations)? Additionally, I can't find any documentation in the main repo, migration guide, or elsewhere that cites this change.

## Script

```python
import hoomd
device = hoomd.device.GPU()

# ...
# rest of simulation
# ...
```

## Output

```
Traceback (most recent call last):
  File "test.py", line 2, in <module>
    device = hoomd.device.GPU()
  File ".../.venv/lib/python3.8/site-packages/hoomd/device.py", line 214, in __init__
    self._cpp_exec_conf = _hoomd.ExecutionConfiguration(
RuntimeError: No supported GPUs are present on this system.
The device Tesla K80 with computed capability 3.7 does not support HOOMD-blue.
```

## Expected output

The K80 would be detected as a supported device.

## Configuration

Platform:
- Linux
- GPU

Installation method:
- Compiled from source

Versions:
- Python version: 3.8
- **HOOMD-blue** version: 3.0-beta10

### PR DESCRIPTION
**Pull request by joaander**
## Description

Inform users what hardware HOOMD-blue developers will make efforts to support in new releases.

## Motivation and context

Some users were not aware that HOOMD-blue does not support hardware that most institutional compute centers deem obsolete.

Resolves #1169

## How has this been tested?

I viewed `ARCHITECTURE.md` in a Markdown previewer.

## Change log

```
Changed:

* Improved documentation.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -11,18 +11,23 @@ our [API documentation][hoomd_documentation] on readthedocs.
 
 [hoomd_documentation]: https://hoomd-blue.readthedocs.io/en/latest/
 
+## Hardware and software support
+
+Each minor and major release of HOOMD-blue at a minimum supports:
+
+* x86_64 CPUs released in the four prior years.
+* NVIDIA GPUs released in the four prior years.
```
Comment (tommy-waltmann) [inline ARCHITECTURE.md:19]: Can we explicitly state what compute capabilities those are?

Comment (joaander) [reply ARCHITECTURE.md:19]: We could, but then we would have to update this document regularly.

Comment (tommy-waltmann) [reply ARCHITECTURE.md:19]: Perhaps we could just state the minimum compute capability we are going to support? Then it wouldn't need to be updated nearly as much.

Comment (joaander) [reply ARCHITECTURE.md:19]: Documenting changes in the minimum required compute capability would be more visible in the change log, along with the other mentions of support removed or added for compilers and libraries.

Comment (joaander) [reply ARCHITECTURE.md:19]: I retroactively added this to the 3.0.0-beta.1 change log.

**Review by tommy-waltmann — APPROVED** [review]

**Review by jekw90 — APPROVED** [review]
The document looks good to me!

**Pull request merged**